packages/ui/src/components/common/TransactionModal/steps/WaitingForDepositStep.tsx

```diff
@@ -0,0 +1,272 @@
+                height: 33,
+                width: 33,
+                excavate: true
+              }}
+              style={{ margin: '0 auto', width: 105, height: 105 }}
+            />
+          </Flex>
+        </div>
+      ) : null}
+      <Pill
+        color="gray"
+        css={{ display: 'flex', alignItems: 'center', gap: '3', p: '3' }}
+      >
+        {isFetchingQuote ? (
+          <Skeleton css={{ height: 16, width: '100%' }} />
+        ) : (
+          <Text style="subtitle2" ellipsify>
+            {depositAddress}
+          </Text>
+        )}
+        <CopyToClipBoard text={depositAddress ?? ''} />
+        <Anchor
+          href={`${blockExplorerBase}/address/${depositAddress}`}
+          target="_blank"
+          color="gray"
+          css={{ display: 'flex', alignItems: 'center', gap: '1' }}
+        >
+          <FontAwesomeIcon
+            icon={faArrowUpRightFromSquare}
+            style={{ width: 16, height: 16 }}
+          />
+        </Anchor>
+      </Pill>
+      <Pill color="amber" css={{ padding: '12px 32px' }} radius="squared">
+        <Text color="warning" style="body3">
+          Don't send {fromToken?.symbol} on any other network or it may be lost
+        </Text>
+      </Pill>
+      <Flex
+        direction="column"
+        align="center"
+        css={{
+          '--borderColor': 'colors.subtle-border-color',
+          border: '1px solid var(--borderColor)',
+          borderRadius: 12,
+          p: '3',
+          gap: '3',
+          width: '100%'
+        }}
+      >
+        <Flex
+          justify="between"
+          align="center"
+          css={{ width: '100%', gap: '4' }}
+        >
+          <Text style="subtitle2" color="subtle">
+            You Get
+          </Text>
+          {isFetchingQuote ? (
+            <Skeleton css={{ width: 80, height: 21 }} />
+          ) : (
+            <Flex
+              align="center"
+              css={{
+                gap: '2'
+              }}
+            >
+              <Text style="subtitle2">~ {toAmountFormatted}</Text>
+              <Pill
+                color="gray"
+                css={{ display: 'flex', alignItems: 'center', gap: '1' }}
+              >
+                <img
+                  alt={toToken?.name}
+                  src={toToken?.logoURI}
+                  width={20}
+                  height={20}
+                  style={{
+                    borderRadius: 9999
+                  }}
+                />{' '}
+                <Text style="subtitle2">{toToken?.symbol}</Text>
+              </Pill>
+            </Flex>
+          )}
+        </Flex>
+
+        <Flex
+          justify="between"
+          align="center"
+          css={{ width: '100%', gap: '4' }}
+        >
+          <Text style="subtitle2" color="subtle">
+            Recipient Wallet
+          </Text>
+          <Text style="subtitle2">{truncateAddress(recipientAddress)}</Text>
```

Comment:
Maybe we should also link out to recipient address and refund wallet address on the block explorer?